trace-normalization/src/normalization/normalizer.rs

```diff
@@ -0,0 +1,100 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0. This product includes software
+// developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present
+// Datadog, Inc.
+
+use std::time::{SystemTime};
+use crate::errors;
+use crate::normalize_utils;
+use crate::pb;
+
+const MAX_TYPE_LEN: usize = 100;
+
+// an arbitrary cutoff to spot weird-looking values
+// nanoseconds since epoch on Jan 1, 2000
+const YEAR_2000_NANOSEC_TS: i64 = 946684800000000000;
+
+// DEFAULT_SPAN_NAME is the default name we assign a span if it's missing and we have no reasonable fallback
+pub const DEFAULT_SPAN_NAME: &str = "unnamed_operation";
+
+#[allow(dead_code)]
+pub fn normalize(s: &mut pb::Span) -> Result<(), errors::NormalizerError> {
+    if s.trace_id == 0 {
+        return Err(errors::NormalizerError::new("TraceID is zero (reason:trace_id_zero)"));
+    }
+    if s.span_id == 0 {
+        return Err(errors::NormalizerError::new("SpanID is zero (reason:span_id_zero)"));
+    }
```

Comment:
I'm not sure if anyone on your team has opinions, but I would recommend pulling in the `anyhow` crate instead of rolling your own errors. It has some nice macros to simplify code around error paths. For instance, the `anyhow::ensure!` macro would reduce the code to something like this:

```rust
pub fn normalize(s: &mut pb::Span) -> anyhow::Result<()> {
    anyhow::ensure!(s.trace_id != 0, "TraceID is zero (reason:trace_id_zero)");
    anyhow::ensure!(s.span_id != 0, "SpanID is zero (reason:span_id_zero)");
```

Reply:
this makes sense, I'll make these changes

Reply:
@DarcyRaynerDD I hadn't heard of Snafu until you mention it, so that pretty much sums up why it wasn't considered :)